peers/database.go

```
@@ -0,0 +1,188 @@
+			if err != nil {
+				return err
+			}
+		}
+		err = savedDb.ForEach(func(k, v []byte) error {
+			addr, _ := multiaddr.NewMultiaddrBytes(k)
+			parsedPeer, err := peer.AddrInfoFromP2pAddr(addr)
+			if err == nil {
+				if parsedPeer.ID == id {
+					// reduce score. If it reaches 0, ban
+					score, _ := strconv.Atoi(string(v))
+					fmt.Printf("peer %s has a banscore of: %s", id.String(), strconv.Itoa(score))
+					score -= 1
+					if score == 0 {
+
+						// add to banlist
+						byteId, err := parsedPeer.ID.MarshalBinary()
+						if err == nil {
+							_ = bansDb.Put(byteId, nil)
+						}
+						// remove from saved list
+						err = savedDb.Delete(k)
+					} else {
+						_ = savedDb.Put(k, []byte(strconv.Itoa(score)))
+					}
+				}
+			}
+			return nil
+		})
+		return err
+	})
+	return err
+}
+
+func IsPeerBanned(netDB *bbolt.DB, id peer.ID) bool {
+	var res bool
+	_ = netDB.Update(func(tx *bbolt.Tx) error {
+		var err error
+		b := tx.Bucket(configBucketKey).Bucket(bansDbKey)
+		if b == nil {
+			b, err = tx.Bucket(configBucketKey).CreateBucketIfNotExists(bansDbKey)
+			if err != nil {
+				return err
+			}
+		}
+		byteId, err := id.MarshalBinary()
+		if err != nil {
+			return err
+		}
+		bannedPeerId := b.Get(byteId)
+		if bannedPeerId != nil {
+			res = true
+			return nil
+		}
+		return nil
+	})
+	return res
+}
+
+func GetSavedPeers(netDB *bbolt.DB) (savedAddresses []multiaddr.Multiaddr, err error) {
+	//retrieve the saved addresses
+	err = netDB.Update(func(tx *bbolt.Tx) error {
+		b := tx.Bucket(configBucketKey).Bucket(peersDbKey)
+		if b == nil {
+			b, err = tx.Bucket(configBucketKey).CreateBucketIfNotExists(peersDbKey)
+			if err != nil {
+				return err
+			}
+		}
+		bannedP := tx.Bucket(configBucketKey).Bucket(bansDbKey)
+		if bannedP == nil {
+			bannedP, err = tx.Bucket(configBucketKey).CreateBucketIfNotExists(bansDbKey)
+			if err != nil {
+				return err
+			}
+		}
+		_ = b.ForEach(func(k, v []byte) error {
```

Don't ignore the error.